src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -124,33 +125,32 @@ private OverdueFineCalculatorService(Clients clients) {
 
   private CompletableFuture<Result<CalculationParameters>> lookupItemRelatedRecords(
     CalculationParameters params) {
+
+    if (params.feeFine == null) {
+      return completedFuture(succeeded(params));
+    }
+
     return itemRepository.fetchItemRelatedRecords(succeeded(params.loan.getItem()))
-      .thenApply(ResultBinding.mapResult(params::withItem));
+      .thenApply(mapResult(params::withItem));
   }
 
   private CompletableFuture<Result<CalculationParameters>> lookupFeeFineOwner(
     CalculationParameters params) {
-    return feeFineOwnerRepository.getFeeFineOwner(
-      params.item.getLocation().getPrimaryServicePointId().toString())
-      .thenApply(ResultBinding.mapResult(params::withOwner));
+
+    return Optional.ofNullable(params.item)
+      .map(Item::getLocation)
+      .map(Location::getPrimaryServicePointId)
+      .map(UUID::toString)
+      .map(id -> feeFineOwnerRepository.getFeeFineOwner(id)
+        .thenApply(mapResult(params::withOwner)))
+      .orElse(completedFuture(succeeded(params)));
```

Comment:
This suggests that if the owner lookup fails, then the process should continue. Is that a reasonable understanding?

If the owner is not found, can a fine be issued?

Reply:
You did understand it right. And no, a fine should not be issued in this case. During the next step we check if all the required entities were fetched successfully, and if they were not, we abort the whole fine creation process.

Reply:
> You did understand it right. And no, a fine should not be issued in this case. During the next step we check if all the required entities were fetched successfully, and if they were not, we abort the whole fine creation process.

Does the user find out that a fine wasn't issued, or does this happen silently?

How are users expected to compensate for this happening? Or is it the correct behaviour that if the type hasn't been set up, then no fines should be issued even if the policy says they should be?

Reply:
Yes, currently this erroneous case is ignored silently. There will be a separate story to handle it.

Reply:
@OleksandrVidinieiev Ok, thanks. 

If that story isn't scheduled for 2020 Q1, please can we make sure folks are aware of this limitation in the release notes for the 2020 Q1 release.

Reply:
@marc-johnson, sure, thanks.